fsm/src/main/java/org/egov/fsm/service/EnrichmentService.java

```diff
@@ -0,0 +1,242 @@
+		}
+		
+		setIdgenIds(fsmRequest);
+		
+	}
+	
+	/**
+	 *  generate the applicationNo using the idGen serivce and populate
+	 * @param request
+	 */
+	private void setIdgenIds(FSMRequest request) {
+		RequestInfo requestInfo = request.getRequestInfo();
+		String tenantId = request.getFsm().getTenantId();
+		FSM fsm = request.getFsm();
+
+		List<String> applicationNumbers = getIdList(requestInfo, tenantId, config.getApplicationNoIdgenName(),
+				config.getApplicationNoIdgenFormat(), 1);
+		ListIterator<String> itr = applicationNumbers.listIterator();
+
+		Map<String, String> errorMap = new HashMap<>();
+
+		if (!errorMap.isEmpty())
+			throw new CustomException(errorMap);
+
+		fsm.setApplicationNo(itr.next());
+	}
+	
+	/**
+	 * Generate the id
+	 * @param requestInfo
+	 * @param tenantId
+	 * @param idKey
+	 * @param idformat
+	 * @param count
+	 * @return
+	 */
+	private List<String> getIdList(RequestInfo requestInfo, String tenantId, String idKey, String idformat, int count) {
+		List<IdResponse> idResponses = idGenRepository.getId(requestInfo, tenantId, idKey, idformat, count)
+				.getIdResponses();
+
+		if (CollectionUtils.isEmpty(idResponses))
+			throw new CustomException(FSMErrorConstants.IDGEN_ERROR, "No ids returned from idgen Service");
+
+		return idResponses.stream().map(IdResponse::getId).collect(Collectors.toList());
+	}
+	
+	/**
+	 *  enrich the update request with the requied ata
+	 * @param fsmRequest
+	 * @param mdmsData
+	 */
+	public void enrichFSMUpdateRequest(FSMRequest fsmRequest, Object mdmsData, FSM oldFsm) {
+		//TODO add requied logic
+		RequestInfo requestInfo = fsmRequest.getRequestInfo();
+		AuditDetails auditDetails = fsmUtil.getAuditDetails(requestInfo.getUserInfo().getUuid(), false);
+		auditDetails.setCreatedBy(oldFsm.getAuditDetails().getCreatedBy());
+		auditDetails.setCreatedTime(oldFsm.getAuditDetails().getCreatedTime());
+		fsmRequest.getFsm().setAuditDetails(auditDetails);
+		if (fsmRequest.getFsm().getAddress() != null) {
+			if (StringUtils.isEmpty(fsmRequest.getFsm().getAddress().getId()))
+				fsmRequest.getFsm().getAddress().setId(UUID.randomUUID().toString());
+			fsmRequest.getFsm().getAddress().setTenantId(fsmRequest.getFsm().getTenantId());
+			fsmRequest.getFsm().getAddress().setAuditDetails(auditDetails);
+			if (fsmRequest.getFsm().getAddress().getGeoLocation() != null
+					&& StringUtils.isEmpty(fsmRequest.getFsm().getAddress().getGeoLocation().getId()))
+				fsmRequest.getFsm().getAddress().getGeoLocation().setId(UUID.randomUUID().toString());
+		}else {
+			throw new CustomException(FSMErrorConstants.INVALID_ADDRES," Address is mandatory");
+		}
+		
+		if(fsmRequest.getFsm().getPitDetail() != null) {
+			if (StringUtils.isEmpty(fsmRequest.getFsm().getPitDetail().getId()))
+				fsmRequest.getFsm().getPitDetail().setId(UUID.randomUUID().toString());
+			fsmRequest.getFsm().getPitDetail().setTenantId(fsmRequest.getFsm().getTenantId());

```

Comment:
enrich tenantId in the if condition, no need to enrich for existing PitDetails

Reply:
Done